### DISCUSSION TITLE
Setting up Strongswan to connect to Fortigate tunnel with specific peer ID

### DISCUSSION THREAD

**Discussion by PaizoSysadmin** [strongswan/strongswan#2572] [General]
I am testing secondary tunnel on my Fortigate firewall to be able to accommodate Linux remote users.  I have set up Ubuntu 22.04 with Strongswan config, but when I try to connect to the VPN tunnel, it just times out.   One thing I am having difficulty finding is how to configure the Fortigate tunnel peer ID in the Strongswan config.  I use a PSK with username and password as well.   We are using IKEv1 in Aggressive mode.

_Here is my ipsec.conf:_

config setup
        strictcrlpolicy=no
        uniqueids = yes
        cachecrls=no
        charondebug="ike 2, knl 2, cfg 2, net 2, esp 2, dmn 2, mgr 2"

conn IPSecTest
        keyexchange=ikev1
        ikelifetime=1440m
        keylife=60m
        aggressive=yes
        ike=aes256-sha256-modp1536,aes256-sha256-modp2048
        mobike=yes
        leftikeport=4500
        rightikeport=4500
        esp=aes256-sha256
        xauth=client
        leftgroups=IPSec-Linux                            _<==== this is the peer id from the fortigate I need to connect to_                
        left=%defaultroute
        leftsourceip=%config
        leftauth=psk
        rightauth=psk
        right=_fortigate ip_
        rightsubnet=_remote subnet_
        xauth_identity="_username_"
        auto=add
        forceencaps=yes

_ipsec statusall:_

Status of IKE charon daemon (strongSwan 5.9.5, Linux 5.15.0-127-generic, x86_64):
  uptime: 55 minutes, since Dec 18 20:03:35 2024
  malloc: sbrk 3203072, mmap 0, used 1201952, free 2001120
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 0
  loaded plugins: charon aesni aes rc2 sha2 sha1 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl fips-prf gmp agent xcbc hmac gcm drbg attr kernel-netlink resolve socket-default connmark stroke updown eap-mschapv2 xauth-generic counters
Listening IP addresses:
  _my local ip_
Connections:
   IPSecTest:  %any..._fortigate ip_ IKEv1 Aggressive
   IPSecTest:   local:  uses pre-shared key authentication
   IPSecTest:    group: IPSec-Linux                                              _<==== this is the peer id from the fortigate I need to connect to_
   IPSecTest:   remote: [_fortigate ip_] uses pre-shared key authentication
   IPSecTest:   child:  dynamic === _remote subnet_ TUNNEL
Security Associations (0 up, 0 connecting):
  none

_this is what i see when i do the ipsec up :_

Initiating Aggressive Mode IKE_SA IPSecTest[1] to _fortigate ip_
generating AGGRESSIVE request 0 [ SA KE No ID V V V V V ]
sending packet: from _my local ip_[4500] to _fortigate ip_[4500] (536 bytes)
sending retransmit 1 of request message ID 0, seq 1
sending packet: from _my local ip_[4500] to _fortigate ip_[4500] (536 bytes)
sending retransmit 2 of request message ID 0, seq 1
sending packet: from _my local ip_[4500] to _fortigate ip_[4500] (536 bytes)

Any guidance that can be provided would be greatly appreciated !

thanks,
Whitney

**Comment by tobiasbrunner** [comment]
> leftgroups=IPSec-Linux _<==== this is the peer id from the fortigate I need to connect to_

This setting is not for that purpose (please [read the docs](https://wiki.strongswan.org/projects/strongswan/wiki/Connsection)). You have to configure that identity in `leftid` instead.

**Comment by PaizoSysadmin** [comment]
Hello Tobias,

 
I did eventually get to this same conclusion working through this issue over the last week or so.   I am now able to establish a connection to my fortigate device.  Now only some slight network tweaks are needed to get DNS and internal/external connectivity working and all will be well.

 
 
Thank you for your response.

 
 
Whitney Chatterjee

System Administrator

Paizo, Inc

 
 
 
From: Tobias Brunner ***@***.***>
Reply-To: strongswan/strongswan ***@***.***>
Date: Monday, January 6, 2025 at 2:11 AM
To: strongswan/strongswan ***@***.***>
Cc: Systems Administrator ***@***.***>, Author ***@***.***>
Subject: Re: [strongswan/strongswan] Setting up Strongswan to connect to Fortigate tunnel with specific peer ID (Discussion #2572)

 
leftgroups=IPSec-Linux <==== this is the peer id from the fortigate I need to connect to

This setting is not for that purpose (please read the docs <https://wiki.strongswan.org/projects/strongswan/wiki/Connsection> ). You have to configure that identity in leftid instead.
